sochain/src/endpoint/balance.ts

```diff
@@ -0,0 +1,106 @@
+import objectPath from 'object-path'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { AdapterRequest } from '@chainlink/types'
+import { Config, DEFAULT_CONFIRMATIONS, DEFAULT_DATA_PATH, getBaseURL } from '../config'
+import { CoinType, isCoinType, ChainType, isChainType } from '.'
+
+export const Name = 'balance'
+
+type Address = {
+  address: string
+  coin?: CoinType
+  chain?: ChainType
+  balance?: number
+}
+
+type RequestData = {
+  dataPath: string
+  confirmations: number
+}
+
+type ResponseWithResult = {
+  response?: any
+  result?: Address
+}
+
+const WARNING_NO_OPERATION_COIN = 'No Operation: unsupported coin type'
+const WARNING_NO_OPERATION_CHAIN = 'No Operation: unsupported chain type'
+const WARNING_NO_OPERATION_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+const getBalanceURI = (network: string, address: string, confirmations: number, chain: string) => {
+  network = network.toUpperCase()
+  if (chain === 'testnet') network = network + 'TEST'
+  return `/api/v2/get_address_balance/${network}/${address}/${confirmations}`
+}
+
+const toBalances = async (
+  jobRunID: string,
+  config: Config,
+  addresses: Address[],
+  confirmations: number = DEFAULT_CONFIRMATIONS,
+): Promise<ResponseWithResult[]> =>
+  Promise.all(
+    addresses.map(async (addr: Address) => {
+      if (!addr.address) return { ...addr, warning: WARNING_NO_OPERATION_MISSING_ADDRESS }
+
+      if (!addr.coin) addr.coin = 'btc'
+      if (isCoinType(addr.coin) === false)
+        throw Requester.errored(jobRunID, WARNING_NO_OPERATION_COIN, 400)
+      if (!addr.chain) addr.chain = 'mainnet'
+      if (isChainType(addr.chain) === false)
+        throw Requester.errored(jobRunID, WARNING_NO_OPERATION_CHAIN, 400)
+
+      const reqConfig = {
+        ...config.api,
+        baseURL: getBaseURL(addr.chain),
+        url: getBalanceURI(addr.coin, addr.address, confirmations, addr.chain),
+      }
+
+      const response = await Requester.request(reqConfig)
+
+      return {
+        response: response.data,
+        result: { ...addr, balance: response.data.data.confirmed_balance },
+      }
+    }),
+  )
+
+const reduceResponse = (responses: ResponseWithResult[]) =>
+  responses.reduce(
+    (accumulator, current) => {
+      accumulator.data.responses = [...accumulator.data.responses, current.response]
+      accumulator.data.result = [...accumulator.data.result, current.result]
+      return accumulator
+    },
+    {
+      data: { responses: [], result: [] },
+      status: 200,
+    } as any,
```

Comment:
`status` is not relevant at this stage, and please type this new type of response.

Reply:
It is relevant at this stage because the output of `reduceResponse` is passed to `Requester.success( )`, which looks for ```
    return {
      jobRunID,
      data: response.data,
      result: response.data.result,
      statusCode: response.status,
    }
```